src/dispatchers/mouseDispatcher.ts

```diff
@@ -75,13 +102,26 @@ export module Dispatcher {
       return this;
     }
 
-    private _processMoveEvent(e: MouseEvent) {
+    public onMouseDown(key: any, callback: MouseCallback): Dispatcher.Mouse {
+      this._setCallback(this._downBroadcaster, key, callback);
+      return this;
+    }
+
+    public onMouseUp(key: any, callback: MouseCallback): Dispatcher.Mouse {
+      this._setCallback(this._upBroadcaster, key, callback);
+      return this;
+    }
+
+    /**
+     * Processes a MouseEvent. Returns false if measurement fails.
+     */
+    private _processMouseEvent(e: MouseEvent) {
       var newMousePosition = this._computeMousePosition(e.clientX, e.clientY);
       if (newMousePosition == null) {
-        return; // couldn't measure
+        return false; // couldn't measure
       }
       this._lastMousePosition = newMousePosition;
-      this._moveBroadcaster.broadcast();
+      return true;
```

Comment:
May want to split this piece of logic up.  Now the name is not obvious stating what this function does especially since it returns a boolean.